src/NHibernate/Impl/StatelessSessionImpl.cs

```diff
@@ -251,25 +245,21 @@ public override void AfterTransactionCompletion(bool successful, ITransaction tx
 
 		public override object GetContextEntityIdentifier(object obj)
 		{
-			CheckAndUpdateSessionStatus();
 			return null;
```

Comment:
I am not seing the point of such a check for yielding `null`. May revert if someone see it better.